src/lib/mongoBackend/mongoUpdateContextSubscription.cpp

```diff
@@ -424,8 +425,26 @@ HttpStatusCode mongoUpdateContextSubscription
     }
   }
 
-  /* Adding format to use in notifications */
-  newSub.append(CSUB_FORMAT, std::string(formatToString(JSON)));
+
+  //
+  // Adding format to use in notifications
+  //
+  std::string         notifyFormatString  = "";
+  NotificationFormat  notifyFormat        = NGSI_NO_NOTIFICATION_FORMAT;
+
+  if (requestP->attrsFormat != NGSI_NO_NOTIFICATION_FORMAT)
+  {
+    notifyFormat       = requestP->attrsFormat;
+    notifyFormatString = notificationFormatToString(notifyFormat);
+    newSub.append(CSUB_FORMAT, notifyFormatString.c_str());
+  }
+  else
+  {
+    notifyFormatString = sub.hasField(CSUB_FORMAT)? getStringFieldF(sub, CSUB_FORMAT) : "JSON";  // If not in database, "JSON" is the default value
```

Comment:
"JSON" -> notificationFormatToString(NGSI_V1_FORMAT)

(Maybe "safer" and more manteinable code in the case we change the keyword in the future).